src/indieweb_utils/replies/context.py

```diff
@@ -7,320 +9,302 @@
-                    author_image = url.split("/")[0] + "//" + domain + author_image
-
-            if h_entry["properties"].get("content") and h_entry["properties"].get("content")[0].get("html"):
-                post_body = h_entry["properties"]["content"][0]["html"]
-                soup = BeautifulSoup(post_body, "html.parser")
-                post_body = soup.text
-
-                favicon = soup.find("link", rel="icon")
-
-                if favicon and not author_image:
-                    photo_url = favicon["href"]
-                    if not photo_url.startswith("https://") or not photo_url.startswith("http://"):
-                        author_image = "https://" + domain + photo_url
-                else:
-                    author_image = None
-
-                post_body = " ".join(post_body.split(" ")[:75]) + " ..."
-            elif h_entry["properties"].get("content"):
-                post_body = h_entry["properties"]["content"]
-
-                post_body = " ".join(post_body.split(" ")[:75]) + " ..."
+    photo: str
+    name: str
+    video: str
+    post_html: str
+    post_text: str
+    authors: List[PostAuthor]
+
+
+class ReplyContextRetrievalError(Exception):
+    def __init__(self, message):
+        self.message = message
+
+class UnsupportedScheme(Exception):
+    def __init__(self, message):
+        self.message = message
+
+def _generate_h_entry_reply_context(h_entry: dict, url: str, parsed_url: str, domain: str, webmention_endpoint_url: str) -> ReplyContext:
+    author_url = ""
+    author_name = ""
+    author_image = ""
+
+    if h_entry["properties"].get("author"):
+        if type(h_entry["properties"]["author"][0]) == dict and h_entry["properties"]["author"][0].get(
+            "type"
+        ) == ["h-card"]:
+            if h_entry["properties"]["author"][0]["properties"].get("url"):
+                author_url = h_entry["properties"]["author"][0]["properties"]["url"][0]
             else:
-                post_body = None
+                author_url = url
 
-            # get p-name
-            if h_entry["properties"].get("name"):
-                p_name = h_entry["properties"]["name"][0]
+            if h_entry["properties"]["author"][0]["properties"].get("name"):
+                author_name = h_entry["properties"]["author"][0]["properties"]["name"][0]
             else:
-                p_name = None
-
-            if author_url is not None and (
-                not author_url.startswith("https://") and not author_url.startswith("http://")
-            ):
-                author_url = "https://" + author_url
-
-            if not author_name and author_url:
-                author_name = url_parse.urlsplit(author_url).netloc
-
-            post_photo_url = None
-            post_video_url = None
-
-            if h_entry["properties"].get("photo"):
-                post_photo_url = canonicalize_url(h_entry["properties"]["photo"][0], domain, url)
+                author_name = ""
```

Comment:
Since author_name is defined at the top of the function, probably don't need to set it blank here, too?